.gitignore

```diff
@@ -9,6 +9,7 @@
 .clangd
```

Comment:
This could all be moved to a separate PR, but not objecting to the changes here either.

Reply:
https://github.com/eth-cscs/DLA-Future/pull/1215

Thanks for the quick merge @rasolca!